### PR TITLE
updates to make service aware of schemaVersion

### PR DESCRIPTION
**Pull request by jh-bate**
@darinkrauss I think I could have improved the config setup a little so open to critic but here is the change with updated tests inline with https://github.com/tidepool-org/tide-whisperer/pull/14

### CODE REVIEW

File: octopus.go [hunk 1]
```diff
@@ -18,9 +18,15 @@ not, you can obtain one from Tidepool Project at tidepool.org.
 package main
 
 import (
+	"crypto/tls"
```
Comment (darinkrauss) [inline octopus.go]: Good cleanup.  We should standardize on putting standard libraries first in import.  Though, does go fmt mess with this?

File: clients/mongoStoreClient_test.go [hunk 2]
```diff
@@ -290,6 +298,51 @@ func TestGetTimeLastEntryUserAndDevice_NoData(t *testing.T) {
...
+
+	//now only data greater or equal to version 101 matters
+	mc101 := initTestData(t, initConfig(101))
+	resultsV101, _ := mc101.ExecuteQuery(allBasals)
+
+	basalsV101 := []found{}
+	json.Unmarshal(resultsV101, &basalsV101)
+
+	if len(basalsV101) != 0 {
+		t.Fatalf("We should have 4 entries for _schemaVersion >= [%d] but got [%d]", mc101.config.SchemaVersionGte, len(basalsV101))
```
Comment (darinkrauss) [inline clients/mongoStoreClient_test.go]: Random question having absolutely nothing to do with this PR:

What is the intent with putting some log values in square brackets?

**Comment by darinkrauss** [comment]
Looks good, though we'll need to discuss the schema version GTE versus range.

File: octopus.go (hunk 1 above)
Comment (jh-bate) [inline octopus.go]: I am using this [sublime add](https://github.com/DisposaBoy/GoSublime) in that runs a go fmt on save . With my current setup that *seems* to push the putting standard libraries to the top and others listed below

File: clients/mongoStoreClient_test.go (hunk 2 above)
Comment (jh-bate) [inline clients/mongoStoreClient_test.go]: habit :)

**Comment by jh-bate** [comment]
@darinkrauss I have updated the PR to have an upper and lower bounds for the schema version and also some general config cleanup

File: clients/mongoStoreClient.go [hunk 3]
```diff
@@ -39,16 +39,28 @@ const (
...
+	config  *StoreConfig
+}
+
+type StoreConfig struct {
+	Connection    *mongo.Config `json:"mongo"`
+	SchemaVersion `json:"schemaVersion"`
+}
+
+type SchemaVersion struct {
+	Gte int
```
Comment (darinkrauss) [inline clients/mongoStoreClient.go]: Same comment as other PR regarding "minimum" and "maximum".

**Comment by darinkrauss** [comment]
Just one comment, otherwise :+1:

File: clients/mongoStoreClient.go (hunk 3 above)
Comment (jh-bate) [inline clients/mongoStoreClient.go]: yeah agreed - will update and then merge

**Pull request merged**